### PR TITLE
Fix privatemessages variable errors

### PR DESCRIPTION
**Pull request by deepend-tildeclub**
Returning the tabs the way they originally were is proving to be a task. Hopefully this is close enough. Line 29 is the only real change on pmfolder.php and line 122 on addfolder.php.

### CODE REVIEW

File: src/members/include/privatemessages/addfolder.php [hunk 1]
```diff
@@ -119,7 +119,7 @@
 				<table class='formTable'>
 					<tr>
 						<td class='formLabel'>Folder Name:</td>
-						<td class='main'><input type='text' name='foldername' value='".$_POST['foldername']."' class='textBox'></td>
+						<td class='main'><input type='text' name='foldername' value='".(isset($_POST['foldername']) ? $_POST['foldername'] : '')."' class='textBox'></td>
```
Comment (RedDragonWebDesign) [inline src/members/include/privatemessages/addfolder.php:122]: Better as `$_POST['foldername'] ?? ''` I think. Shorter and more readable.

**Review by RedDragonWebDesign — CHANGES_REQUESTED** [review]

File: src/members/include/privatemessages/addfolder.php (hunk 1 above)
Comment (deepend-tildeclub) [reply src/members/include/privatemessages/addfolder.php]: Yeah good point.  I think i have a bunch in another file.. I'll try to adjust those ones too before I resubmit those other patches.. Slowly getting these cleaned up a bit and submitted.  Time consuming cleaning up my mess :)

File: src/members/include/privatemessages/addfolder.php [hunk 2]
```diff
@@ -119,7 +119,7 @@
 				<table class='formTable'>
 					<tr>
 						<td class='formLabel'>Folder Name:</td>
-						<td class='main'><input type='text' name='foldername' value='".$_POST['foldername']."' class='textBox'></td>
+						class='main'><input type='text' name='foldername' value='".$_POST['foldername'] ?? ''."' class='textBox'></td>
```
Comment (RedDragonWebDesign) [inline src/members/include/privatemessages/addfolder.php:122]: Looks like `<td ` was accidentally deleted here. Please re-add.

Comment (RedDragonWebDesign) [inline src/members/include/privatemessages/addfolder.php:122]: I looked up the [PHP order of precedence](https://www.php.net/manual/en/language.operators.precedence.php) of `??`, and it has a lower precedence than `.` So we need to surround this with parentheses. `.($_POST['foldername'] ?? '').`

File: src/classes/pmfolder.php [hunk 3]
```diff
@@ -26,6 +26,7 @@ public function __construct($sqlConnection) {
 		}
 		
 		public function select($intIDNum, $numericIDOnly = true) {
+			$returnVal = false; 
 			$arrSpecialFolders = array("Inbox" => self::INBOX_ID, "Sent Messages" => self::SENTBOX_ID, "Trash" => self::TRASH_ID);
 			if(in_array($intIDNum, $arrSpecialFolders)) {
 				$this->arrObjInfo['name'] = array_search($intIDNum, $arrSpecialFolders);
```
Comment (RedDragonWebDesign) [inline src/classes/pmfolder.php:32]: While we're in here, can you change `numericIDOnly` to `$numericIDOnly` on line 37? That is clearly meant to be a variable not a constant so will probably fix something :)

**Review by RedDragonWebDesign — CHANGES_REQUESTED** [review]

File: src/classes/pmfolder.php (hunk 3 above)
Comment (deepend-tildeclub) [reply src/classes/pmfolder.php:32]: nice catch :)

Comment (RedDragonWebDesign) [reply src/classes/pmfolder.php:32]: My code editor put a red squiggle under it. X-ray vision :)

Comment (deepend-tildeclub) [reply src/classes/pmfolder.php:32]: lol  Do you use VS Code.. If so share with me if you have any certain extensions installed or if its just Vanilla VS Code.

Comment (NovemLinguae) [reply src/classes/pmfolder.php:32]: Yes, VS Code. In this case, the PHP Intelephense extension caught it. I have 30ish extensions.

image

Comment (RedDragonWebDesign) [reply src/classes/pmfolder.php:32]: Yes, VS Code. In this case, the PHP Intelephense extension caught it. I have 30ish extensions.

image

Comment (deepend-tildeclub) [reply src/classes/pmfolder.php:32]: probably why your seeing real obvious stuff that I'm just not noticing while making the changes..  added that extension to mine right now :)

**Comment by RedDragonWebDesign** [comment]
Fixes fatal error on PM Add Folder screen and a warning on the PM homepage. Thanks!

**Pull request merged**